inc/classes/subscriber/Optimization/class-abstract-minify-subscriber.php

```diff
@@ -0,0 +1,197 @@
+<?php
+namespace WP_Rocket\Subscriber\Optimization;
+
+use WP_Rocket\Admin\Options_Data as Options;
+use Wa72\HtmlPageDom\HtmlPageCrawler;
+
+/**
+ * Undocumented class
+ */
+abstract class Minify_Subscriber {
+	/**
+	 * Plugin options
+	 *
+	 * @since 3.1
+	 * @author Remy Perona
+	 *
+	 * @var Options
+	 */
+	protected $options;
+
+	/**
+	 * Crawler instance
+	 *
+	 * @since 3.1
+	 * @author Remy Perona
+	 *
+	 * @var HtmlPageCrawler
+	 */
+	protected $crawler;
+
+	/**
+	 * Optimizer instance
+	 *
+	 * @since 3.1
+	 * @author Remy Perona
+	 *
+	 * @var Optimizer_Interface
+	 */
+	protected $optimizer;
+
+	/**
+	 * Constructor
+	 *
+	 * @since 3.1
+	 * @author Remy Perona
+	 *
+	 * @param Options         $options Plugin options.
+	 * @param HtmlPageCrawler $crawler Crawler instance.
+	 */
+	public function __construct( Options $options, HtmlPageCrawler $crawler ) {
+		$this->options = $options;
+		$this->crawler = $crawler;
+	}
+
+	/**
+	 * Sets the type of optimizer to use
+	 *
+	 * @since 3.1
+	 * @author Remy Perona
+	 *
+	 * @param Optimizer_Interface $optimizer Optimizer instance.
+	 * @return void
+	 */
+	protected function set_optimization_type( $optimizer ) {
+		$this->optimizer = $optimizer;
+	}
+
+	/**
+	 * Processes the HTML to perform an optimization and return the new content
+	 *
+	 * @since 3.1
+	 * @author Remy Perona
+	 *
+	 * @param string $html HTML content.
+	 * @return string
+	 */
+	abstract public function process( $html );
+
+	/**
+	 * Checks if files can be optimized
+	 *
+	 * @since 3.1
+	 * @author Remy Perona
+	 */
+	abstract protected function is_allowed();
+
+	/**
+	 * Performs the optimization
+	 *
+	 * @since 3.1
+	 * @author Remy Perona
+	 *
+	 * @return string
+	 */
+	protected function optimize() {
+		return $this->optimizer->optimize();
+	}
+
+	/**
+	 * Fix issue with SSL and minification
+	 *
+	 * @since 2.3
+	 *
+	 * @param string $url An url to filter to set the scheme to https if needed.
+	 * @return string
+	 */
+	public function fix_ssl_minify( $url ) {
+		if ( ! is_ssl() ) {
+			return $url;
+		}
+
+		if ( false !== strpos( $url, 'https://' ) ) {
```

Comment:
It's kind of double negation no? I may be simpler to understand (and more strict):
```php
if ( 0 === strpos( $url, 'https://' ) ) {
```